upb/json/parser.rl

```diff
@@ -2008,86 +1962,115 @@ static bool is_string_wrapper_object(upb_json_parser *p) {
 static bool does_boolean_wrapper_start(upb_json_parser *p) {
   return p->top->f != NULL &&
          upb_fielddef_issubmsg(p->top->f) &&
-         is_bool_value(upb_fielddef_msgsubdef(p->top->f));
+         (upb_msgdef_wellknowntype(upb_fielddef_msgsubdef(p->top->f))
+              == UPB_WELLKNOWN_BOOLVALUE);
 }
 
 static bool does_boolean_wrapper_end(upb_json_parser *p) {
-  return p->top->m != NULL && is_bool_value(p->top->m);
+  return p->top->m != NULL &&
+         (upb_msgdef_wellknowntype(p->top->m)
+              == UPB_WELLKNOWN_BOOLVALUE);
```

Comment:
A lot of functions are using this pattern. Create a function like this?

```
static bool top_is_wellknown(upb_json_parser *p, upb_wellknown_t type) {
  return p->top->m != NULL & upb_msgdef_wellknowntype(p->top->m) == type;
}
```

Then I don't think you would need all these separate functions.